server/src/main/java/org/opensearch/index/fielddata/IndexNumericFieldData.java

```diff
@@ -133,8 +134,16 @@ public final SortField sortField(
             : SortedNumericSelector.Type.MIN;
         SortField sortField = new SortedNumericSortField(getFieldName(), getNumericType().sortFieldType, reverse, selectorType);
         sortField.setMissingValue(source.missingObject(missingValue, reverse));
+        
+        // For long, doubles & dates, we can use this numeric sort optimiser.
+        boolean enablePointsOptimization = Arrays.asList(NumericType.DATE,
+            NumericType.DATE_NANOSECONDS,
+            NumericType.LONG,
+            NumericType.DOUBLE).contains(getNumericType());
+
         // todo: remove since deprecated
-        sortField.setOptimizeSortWithPoints(false);
+        sortField.setOptimizeSortWithPoints(enablePointsOptimization);
```

Comment:
Why set this at all, as opposed to relying on the default? The Lucene docs say:

> should only be used for compatibility with 8.x indices that got created with inconsistent data across fields, or the wrong sort configuration in the index sort

Is compatibility a concern here? We definitely need to support 8.x indices.

Reply:
We cant do this for all sort Types. It will break the sort functionality. I have tested this enabling this for all types and sorting on SHORT was failing with below exception 
`{
        "type" : "illegal_argument_exception",
        "reason" : "Field size is indexed with 4 bytes per dimension, but org.apache.lucene.search.SortedNumericSortField$3@510750f1 expected 8"
},`

We did widening of Sorting field here which is different of Point type used during indexing.
https://code.amazon.com/packages/OpenSearch/blobs/opensearch-2.3.0/--/opensearch/src/server/src/main/java/org/opensearch/index/fielddata/IndexNumericFieldData.java#:~:text=BOOLEAN(false,.NUMERIC)%3B

Reply:
@gashutos we need to move it to `NumericType` as a property, fe `boolean supportsPointsOptimization()` so each `NumericType` instance could provide these hints.

Reply:
Btw, is it scheduled for removal in Lucene 10

```
  @Deprecated // Remove in Lucene 10
  public boolean getOptimizeSortWithPoints() {
    return getOptimizeSortWithIndexedData();
  }
```

Reply:
I thought of having it in NumericType, but later .
Yes, this is suposed to be removed in Lucene 10. 
So setting this boolean is temporary. That's why I am thinking to keep it at  one place only, so we can remove entire block together later.

Reply:
@reta see my [comment here](https://github.com/opensearch-project/OpenSearch/pull/6321#discussion_r1106707510) about keeping this specific to the logic path that requires it.